h2o-core/src/main/java/water/fvec/NewChunk.java

```diff
@@ -648,16 +648,21 @@ public void addUUID(UUID uuid) {
 
   // Append a UUID, stored in _ls & _ds
   public void addUUID( long lo, long hi ) {
-    if (C16Chunk.isNA(lo, hi)) throw new IllegalArgumentException("Cannot set illegal UUID value");
-    if( _ms==null || _ds== null || _sparseLen >= _ms.len() )
-      append2slowUUID();
-    _ms.set(_sparseLen,lo);
-    _ds[_sparseLen] = Double.longBitsToDouble(hi);
-    if (_id != null) _id[_sparseLen] = _len;
+    set_impl(_sparseLen, lo, hi);
     _sparseLen++;
     _len++;
     assert _sparseLen <= _len;
   }
+
+  @Override public boolean set_impl(int idx, long lo, long hi) {
```

Comment:
"set_impl" is a confusing name for this function. This function append UUID to the chunk - it does't implement random-access set, semantics of set_impl methods RA set to a given position.

I recommend to keep the original function addUUID as is.